vm/vm.go

```diff
@@ -133,12 +133,37 @@ type VM struct {
 	stop  chan struct{}
 }
 
-func New(factory ControllerFactory, v *version.Semantic) *VM {
-	return &VM{
-		factory: factory,
-		v:       v,
-		config:  NewConfig(),
+func New(
+	factory ControllerFactory,
+	v *version.Semantic,
+	actionRegistry chain.ActionRegistry,
+	authRegistry chain.AuthRegistry,
+	authEngine map[uint8]AuthEngine,
+	options ...Option,
+) (*VM, error) {
+	vm := &VM{
+		factory:        factory,
+		v:              v,
+		config:         NewConfig(),
+		actionRegistry: actionRegistry,
+		authRegistry:   authRegistry,
+		authEngine:     authEngine,
+	}
+
+	txGossiper, err := gossiper.NewProposer(vm, gossiper.DefaultProposerConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	// Set defaults
+	vm.builder = builder.NewTime(vm)
+	vm.gossiper = txGossiper
```

Comment:
optional nit to set gossiper right after it's created

```suggestion
	txGossiper, err := gossiper.NewProposer(vm, gossiper.DefaultProposerConfig())
	if err != nil {
		return nil, err
	}
	// Set defaults
	vm.gossiper = txGossiper
	vm.builder = builder.NewTime(vm)
```